scripts/tasks/generate-license-files.js

```diff
@@ -23,14 +25,24 @@ async function exists(filename) {
     }
 }
 
+const licenses = JSON.parse(
+    childProcess.spawnSync('pnpm', ['licenses', 'list', '--json'], { encoding: 'utf-8' }).stdout
+);
+
+if (!licenses || typeof licenses !== 'object') {
+    throw new Error('Failed to list dependencies');
+}
+
+const list = Object.values(licenses).flat();
+
 async function findLicenseText(depName) {
     // Iterate through possible names for the license file
     const names = ['LICENSE', 'LICENSE.md', 'LICENSE.txt'];
 
     // We would use require.resolve, but 1) that doesn't work if the module lacks a "main" in its `package.json`,
     // and 2) it gives us a deep `./path/to/index.js` which makes it harder to find a top-level LICENSE file. So
     // just assume that our deps are hoisted to the top-level `node_modules`.
-    const resolvedDepPath = path.join(process.cwd(), 'node_modules', depName);
+    const resolvedDepPath = list.find((dep) => dep.name === depName).paths[0];
```

Comment:
Why does `generate-license-files.js` need to be updated?

Reply:
The dependencies are not hoisted to the top-level anymore. Transitive dependencies like `entities` from '@parse5/tools` are also not hoisted to the package which depends on it. This is the only reasonable way to accomplish this.

Refs:
- https://pnpm.io/symlinked-node-modules-structure
- https://pnpm.io/how-peers-are-resolved